src/app/shared/search-box/search-box.component.ts

```diff
@@ -127,18 +129,42 @@ export class SearchBoxComponent implements OnInit, OnDestroy {
 
   public onAddQueryItem(queryItem: QueryItem) {
     this.queryItems.push(queryItem);
-    this.queryItemsControl.push(queryItemToForm(queryItem));
+
+    const queryItemForm = queryItemToForm(queryItem);
+    this.queryItemsControl.push(queryItemForm);
+
+    if (queryItemForm.valid) {
+      this.onQueryItemsChanged();
+    }
   }
 
   public onRemoveQueryItem(index: number) {
     this.queryItems.splice(index, 1);
+
+    const isQueryItemValid =  this.queryItemsControl.at(index).valid;
     this.queryItemsControl.removeAt(index);
+
+    if(isQueryItemValid) {
```

Comment:
Why do you check the validity of the query item which is being removed? Does it matter if it was valid or not? I think more important information is if the whole search box form is valid at this point.

Reply:
It is same as above.. thanks for finding that 👍